src/sqlfluff/rules/L045.py

```diff
@@ -44,25 +43,20 @@ class Rule_L045(BaseRule):
     """
 
     @classmethod
-    def _visit_sources(
```

Comment:
This PR is focused on fixing L044, which involved major changes to `SelectCrawler`. Here, we update L045 so it works with the updated `SelectCrawler`. This PR does not address #1999, "L045 should handle nested CTEs".